src/bridge_iroad/src/iroad_closedloop.cpp

```diff
@@ -91,32 +102,46 @@ void Teleop_IRoad::udp_cb(const ros::TimerEvent& event) {
 
   Matrix<double,3,1> r_f;       // Front vector in body frame
   r_f << 0.0,1.0,0.0;
-  steer = rad2deg(acos(r_tb.dot(r_f)/d_t));
+  
+  /*steer = rad2deg(acos(r_tb.dot(r_f)/d_t)); //debug: steering setting is equal to angle between vehicle heading and goal direction
 
-  steer = min(steer, 30.0);
+  steer = min(steer, max_steer);
+  steer /= max_steer;
 
   if (r_tb(0,0) < 0) {
     steer = -steer;
   } else {
     // Carry On
   }
 
+  */
+
+  double alpha = acos(r_tb.dot(r_f)/d_t); //angle between vehicle heading and direction of goal (in radians)
+  if (r_tb(0,0) < 0){
+    alpha = -alpha; //accounting for the difference between left and right turns
+  }
+  //double delta; //angle between vehicle heading and desired vehicle heading at target (in radians)
+
+  steer = rad2deg(atan2(whlbase*(ctrl_k2*alpha + ctrl_k1*sin(alpha)*cos(alpha)),v_const))/max_steer; //calculation of steering without regard for desired heading at target
```

Comment:
split this into smaller equations. i can't quite understand what you're trying to do here.